compiler/rustc_pattern_analysis/src/lib.rs

```diff
@@ -79,6 +79,13 @@ pub struct MatchArm<'p, Cx: MatchCx> {
 
 impl<'p, Cx: MatchCx> Copy for MatchArm<'p, Cx> {}
 
+#[cfg(not(feature = "rustc"))]
+pub trait Captures<'a> {}
+#[cfg(not(feature = "rustc"))]
+impl<'a, T: ?Sized> Captures<'a> for T {}
+#[cfg(feature = "rustc")]
+pub use rustc_data_structures::captures::Captures;
```

Comment:
We should just unconditionally duplicate the `Captures` trait here. No reason to import it from `rustc_data_structures`, it's small enough that code repetition is not an issue.